drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,931 @@
+		SET_MASK(pComPort->UMCR, MCR_AFE | MCR_RTS);
+		pComPort->URTR  = 56;
+	}
+#endif /* DEF_ENABLE_CTS_RTS */
+
+	/* set baudrate */
+#if CONFIG_ITE_UART_DMA_MODE
+	pComPort->UBRCDIR = pBR->baud_int;
+	pComPort->UBRCDDR = pBR->baud_den;
+	pComPort->UBRCDNR = pBR->baud_num;
+#endif /* CONFIG_ITE_UART_DMA_MODE */
+	SET_MASK(pComPort->ULCR, LCR_DLAB);
+	pComPort->UDLL = pBR->baud_udll;
+	pComPort->UDLM = pBR->baud_udlm;
+	pComPort->UFCR = FCR_FIFO_EN | FCR_RFRST | FCR_XFRST |
+					FCR_64B_FIFO_EN | FCR_FIFO_8;
+	pComPort->ULCR = LCR_8N1 | (is_enable_parity ? LCR_PEN : 0) |
+					(is_event_parity ? LCR_EPS : 0) |
+					(is_onestop_bit ? 0 : LCR_STB);
+	ISR4  |= BIT(6 + chan);
+	SET_MASK(pComPort->UMCR, MCR_OUT2);
+}
+
+/**
+ *
+ * get_br_idx
+ *
+ * @return
+ * STATUS_SUCCESS - input baud rate is match supported type
+ * STATUS_WARN_BAUDRATE - input baud rate isn't match, choice the ceiling
+ * @parameter
+ *
+ *
+ */
+static int get_br_idx(uint32_t baud_rate, uint32_t *baud_rate_index)
+{
+	int ret = STATUS_SUCCESS;
+
+	if (baud_rate <= 4800) {
+		*baud_rate_index = t_br_4800;
+		if (baud_rate != 4800) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 9600) && (baud_rate > 4800)) {
+		*baud_rate_index = t_br_9600;
+		if (baud_rate != 9600) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 14400) && (baud_rate > 9600)) {
+		*baud_rate_index = t_br_14400;
+		if (baud_rate != 14400) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 19200) && (baud_rate > 14400)) {
+		*baud_rate_index = t_br_19200;
+		if (baud_rate != 19200) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 38400) && (baud_rate > 19200)) {
+		*baud_rate_index = t_br_38400;
+		if (baud_rate != 38400) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 57600) && (baud_rate > 38400)) {
+		*baud_rate_index = t_br_57600;
+		if (baud_rate != 57600) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 115200) && (baud_rate > 57600)) {
+		*baud_rate_index = t_br_115200;
+		if (baud_rate != 115200) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if ((baud_rate <= 230400) && (baud_rate > 115200)) {
+		*baud_rate_index = t_br_230400;
+		if (baud_rate != 230400) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	} else if (baud_rate > 230400) {
+		/* the max baud rate is 460800 */
+		*baud_rate_index = t_br_460800;
+		if (baud_rate != 460800) {
+			ret = STATUS_WARN_BAUDRATE;
+		}
+	}
+	return ret;
+}
+
+#define DEV_CFG(dev)					      \
+	((const struct uart_ite_it8xxx2_device_config *const) \
+	 (dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct uart_ite_it8xxx2_dev_data_t *)(dev)->driver_data)
+struct uart_ite_it8xxx2_device_config {
+	uint32_t sys_clk_freq;
+};
+/** Device data structure */
+struct uart_ite_it8xxx2_dev_data_t {
+	uint32_t base;
+	uint32_t port;
+	uint32_t baud_rate;     /**< Baud rate */
+	uint32_t irq_no;
+	#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+	/**< cache of IIR read clear */
```

Comment:
//*< seems wrong to me since you are not commenting the thing before. I think it should be /**